src/styles/app-button/AppButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { useButton } from "@react-aria/button";
+import { useRef } from "react";
+
+import classnames from "classnames";
+
+import "@/styles/app-button/AppButton.scss";
+import { AppButtonProps } from "@/styles/app-button/AppButton.types";
+
+const AppButton = function ({
+  variant = "primary",
+  size = "md",
+  full = false,
+  loading = false,
+  disabled = false,
+  inline = false,
+  children,
+  ...props
+}: AppButtonProps) {
+  const ref = useRef<HTMLButtonElement | null>(null);
+  const { buttonProps } = useButton(props, ref);
```

Comment:
Can we destructure and use `isPressed`?

Reply:
![image](https://github.com/user-attachments/assets/afdb3ea0-9713-431f-a08e-f9f75d32b2f5)
https://react-spectrum.adobe.com/react-aria/useButton.html#usebutton